arch/arc/core/thread.c

```diff
@@ -91,19 +90,19 @@ void arch_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 		thread->arch.priv_stack_start =
 			(u32_t)(stackEnd + STACK_GUARD_SIZE);
 
-		stackAdjEnd = (char *)STACK_ROUND_DOWN(stackEnd +
+		stackAdjEnd = (char *)Z_STACK_PTR_ALIGN(stackEnd +
 				ARCH_THREAD_STACK_RESERVED);
 
 		/* reserve 4 bytes for the start of user sp */
 		stackAdjEnd -= 4;
-		(*(u32_t *)stackAdjEnd) = STACK_ROUND_DOWN(
+		(*(u32_t *)stackAdjEnd) = Z_STACK_PTR_ALIGN(
 			(u32_t)stackEnd - offset);
```

Comment:
This is somewhat unrelated to this PR, but I wonder if we should fix this while we are at it (AFAICS, we do not need to do pointer-to-integer conversion here at all).